test/extensions/quic_listeners/quiche/active_quic_listener_test.cc

```diff
@@ -112,23 +113,27 @@ class ActiveQuicListenerTest : public QuicMultiVersionTest {
     ON_CALL(listener_config_, listenSocketFactory()).WillByDefault(ReturnRef(socket_factory_));
     ON_CALL(socket_factory_, getListenSocket()).WillByDefault(Return(listen_socket_));
 
-    // Use UdpGsoBatchWriter to perform non-batched writes for the purpose of this test
+    // Use UdpGsoBatchWriter to perform non-batched writes for the purpose of this test, if it is
+    // supported.
     ON_CALL(listener_config_, udpPacketWriterFactory())
         .WillByDefault(Return(
             std::reference_wrapper<Network::UdpPacketWriterFactory>(udp_packet_writer_factory_)));
     ON_CALL(udp_packet_writer_factory_, createUdpPacketWriter(_, _))
         .WillByDefault(Invoke(
             [&](Network::IoHandle& io_handle, Stats::Scope& scope) -> Network::UdpPacketWriterPtr {
-              Network::UdpPacketWriterPtr udp_packet_writer =
-                  std::make_unique<Quic::UdpGsoBatchWriter>(io_handle, scope);
-              return udp_packet_writer;
+#if UDP_GSO_BATCH_WRITER_PLATFORM_SUPPORT
```

Comment:
This makes the test compile and pass on MacOs. There's another PR out right now that does roughly the same for Windows, and this will make that pass as well (#12898)

Reply:
I wanted to make the logic of whether it is supported be contained to the header that defines it. I would like to change that header to check for the existence of the syscall in question, although I'm not exactly sure how to do that. 

If an implementation becomes available for other operating systems, the definition of the class could be updated without needing to change this test code.

Reply:
> > > > > * Disabling a quic listener wouldn't immediately stop accepting new connections; it only stopped accepting connections that were buffered. The implementation would allow up to 16 connections to be accepted after disabling the listener. This made it tricky to make tests work consistently. Fixed by having `EnvoyQuicDispatcher::ShouldCreateOrBufferPacketForConnection()` check the runtime value.
> > > > 
> > > > 
> > > > Disabling quic listener would immediately stop process any QUIC payload.
> > > 
> > > 
> > > `onData()` was still processing packets, even though `onReadReady()` was not. Or at least one of the tests made it appear that it was still processing; it's possible that the full stack (not just ActiveQuicListener) was disabling all processing. I made that note after fiddling with `active_quic_listener_test`, not an integration test or actual running Envoy.
> > 
> > 
> > Interesting, ActiveQuicListenerTest.QuicProcessingDisabledAndEnabled should be testing that feature.
> 
> Yes, the behavior is different if the listener has a read-event before it gets disabled, because that resulted in a call to `ProcessBufferedChlos(kNumSessionsToCreatePerLoop)`, which would allow the dispatcher to accept `kNumSessionsToCreatePerLoop`, since it was still processing packets in `onData()`. It's a corner case that the test wasn't hitting.

Are you saying the feature flag might be inconsistent between calling `ProcessBufferedChlos()` and `onData()`? `ProcessBufferedChlos()` and `onData()` are called in the same call stack. I'm not familiar with runtime::FeatureFlag. Is it possible for it to change within a call stack?

Anyway, the PR description seems outdated, can you update it?

Reply:
I modified the test here: https://github.com/envoyproxy/envoy/pull/13086/commits/adffbb1a6076612904c7fafe6f01276625252a56

The new version of the test fails on current master, and passes in this PR branch. It's not because of a changed value of FeatureFlag, it's because QuicDispatcher remembers how many CHLOs it can process before buffering, but the value starts at zero when QuicDispatcher is constructed, so the behavior is different between a newly constructed QuicDispatcher, and one that has had ProcessBufferedChlos() called on it (with no CHLOs to process, or fewer than the limit at least).

Reply:
Updated PR description.

Reply:
Right now GSO support is determined at runtime.  But GSO code can also compile at the linux kernels that doesn't support it. I think what we want here is `PLATFORM_SUPPORT_POSIX` or something like that.

Reply:
> I modified the test here: [adffbb1](https://github.com/envoyproxy/envoy/commit/adffbb1a6076612904c7fafe6f01276625252a56)
> 
> The new version of the test fails on current master, and passes in this PR branch. It's not because of a changed value of FeatureFlag, it's because QuicDispatcher remembers how many CHLOs it can process before buffering, but the value starts at zero when QuicDispatcher is constructed, so the behavior is different between a newly constructed QuicDispatcher, and one that has had ProcessBufferedChlos() called on it (with no CHLOs to process, or fewer than the limit at least).

Thanks for the explanation! It makes a lot of sense! Can you modify the test accordingly to prevent regression?

Reply:
That commit is part of this PR, so it's already included.